prefetch.bs

```diff
@@ -67,31 +67,80 @@ spec: RFC8941; urlPrefix: https://www.rfc-editor.org/rfc/rfc8941.html
     text: List; url: name-lists
     text: Token; url: name-tokens
 spec: COOKIES; urlPrefix: https://httpwg.org/specs/rfc6265.html
+  type: http-header; text: Cookie; url: cookie
   type: http-header; text: Set-Cookie; url: set-cookie
   type: dfn; text: cookie; url: storage-model
   type: dfn; text: receive a cookie; url: storage-model
+  type: dfn; text: domain-matches; url: cookie-domain
+  type: dfn; text: canonicalized host name; url: cookie-domain-canonicalize
+  type: dfn; text: path-matches; url: cookie-path
 </pre>
 
 <h2 id="concepts">Concepts</h2>
 
 In light of <a href="https://privacycg.github.io/storage-partitioning/">storage partitioning</a>, this specification defines prefetch for navigations which would occur within the same partition (for example, top-level navigations within the same site) and for navigations which would occur in a separate partition (for example, top-level navigations to a different site).
 
+An <dfn>exchange record</dfn> is a [=struct=] with the following [=struct/items=]:
+* <dfn for="exchange record">request</dfn>, a [=request=]
+* <dfn for="exchange record">response</dfn>, a [=response=] or null
+* <dfn for="exchange record">cookie data</dfn>, a [=list=] of [=tuples=] of [=byte sequences=]
+* <dfn for="exchange record">authentication entry</dfn>, an [=authentication entry=] or null
+
+<div class="note">These records can be used to defer checks that would ordinarily happen during a navigate fetch, and to check for modified [=credentials=].</div>
+
+A <dfn>redirect chain</dfn> is a [=list=] of [=exchange records=].
+
+<div algorithm>
+    To <dfn for="redirect chain">append</dfn> a [=request=] |request| to a [=redirect chain=] |redirectChain|:
+
+    1. Let |cookieData| be the result of [=gathering cookie data=] for |request|'s [=request/URL=] in the cookie store associated with |request| if the user agent is not configured to block cookies for |request|, and an empty [=list=] otherwise.
```

Comment:
Hmm, that seemed a little awkward to me too. Changed to start with an empty list and replace it with the cookies if it can, which seemed clearer and is similar to the following lines.